### ISSUE TITLE
pod-culler upgrade - ServiceAccount required before initialized?

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/zero-to-jupyterhub-k8s#619]
I toss this issue out before I go to bed, got any suggestions I'm happy to hear about them. I'll spend more time figuring things out later.

---

I just attempted to update to a more recent release of this repo, but got the following error.

```Error: UPGRADE FAILED: no ServiceAccount with the name "pod-culler" found```

It makes little sense to me, I can see the rbac.yaml definition within pod-culler template directory. I think my deployment is conditioned out by my `cull.maxAge` being 0.

```yaml
# my values.yaml
rbac:
  enabled: true

cull:
  enabled: true
  users: false
  timeout: 3600
  every: 600
  maxAge: 0

proxy:
  https:
    enabled: true
    type: letsencrypt
    letsencrypt:
      contactEmail: 'myemail@host.com'
    hosts: [mydomain.se]
```

https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/d7d7a58636b9556297cd2f71fe7f077e10534025/jupyterhub/templates/pod-culler/deployment.yaml#L1-L5

https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/d7d7a58636b9556297cd2f71fe7f077e10534025/jupyterhub/templates/pod-culler/rbac.yaml#L1-L5

**Comment by consideRatio** [comment]
OH! perhaps we need to move the ServiceAccount from the bottom of the rbac.yaml to the top?

**Comment by consideRatio** [comment]
### Notes
- `cull.maxAge: 0`, but the conditional `{{ if and .Values.cull.enabled .Values.cull.maxAge }}` seem to be evaluated to true (ping @choldgraf --- something of relevance? I recall you had issues with the culler.)
- I had to delete a serviceaccount before it could be recreated again to get rid of the "ServiceAccount not found" trouble.
- Gaaaah whats going on with the "no .... of the name .... found" ?!?!??!!? I figure I need to cleanup helms release number history or similar...

**Comment by consideRatio** [comment]
Resolved --- By using helm 2.8.2 with a clean slate. To learn more see #620
